src/components/Documentation/Markdown/index.tsx

```diff
@@ -30,17 +36,31 @@ const isInsideCodeBlock = (node: Element): boolean => {
 }
 
 const Details: React.FC<{
-  children: Array<{ props: { children: Array<string> } } | string>
+  children: Array<{ props: { children: ReactNode } } | string>
 }> = ({ children }) => {
   const filteredChildren = children.filter(child => child !== '\n')
 
   if (!filteredChildren.length) return null
   if (typeof filteredChildren[0] === 'string') return null
 
-  const text = filteredChildren[0].props.children[0]
+  const headingChildren: ReactNode[] = filteredChildren[0].props
+    .children as ReactNode[]
+
+  // Remove header auto-link if present
+  const finalHeadingChild = headingChildren[headingChildren.length - 1] as {
+    props: {
+      className: string
+    }
+  }
+
+  const triggerChildren: unknown =
```

Comment:
is it actually `unknown`?

Reply:
Kind of. It should be a `ReactNode`, but since `Collapsible`'s interface for the `trigger` prop only accepts `text` or `ReactElement` despite `ReactNode`s being a viable input.

Because of this, I TS casted `triggerChildren` to `unknown` to "force" a cast from `ReactNode` to `ReactElement` at the suggestion of the compiler.
Maybe making this a `ReactNode` and doing `as unknown as ReactComponent` would be a little more expressive, so readers know exactly what we're doing.

Reply:
can we do `ReactNode` here, and  `as ReactElement ` below? (I'm not very familiar with TS rules tbh, especially when it comes to TS being passed to JS + PropTypes, just curious how this is handled usually). Btw, does TS "understand" propTypes?

Reply:
Because of a quirk of how this conversion works with raw children, the closest we can get is `ReactNode` here, and `as unknown as ReactElement` below. Since `ReactNode`s don't cleanly translate into `ReactComponent`, the variable has to pass through `unknown` at some point to let the compiler know it isn't a mistake.

Reply:
👍